cmake/CompilerHelper.cmake

```diff
@@ -1,7 +1,7 @@
 # =============================================================================
 # Compiler specific settings
 # =============================================================================
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR NRN_MACOS_BUILD)
   set(UNDEFINED_SYMBOLS_IGNORE_FLAG "-undefined dynamic_lookup")
```

Comment:
I feel bad about missing this. It was silly of me to think that intel icc uses a special intel linker and not the normal mac version.